openbb_platform/providers/fred/openbb_fred/models/federal_funds_rate.py

```diff
@@ -0,0 +1,205 @@
+            wef = Weekly, Ending Friday
+            weth = Weekly, Ending Thursday
+            wew = Weekly, Ending Wednesday
+            wetu = Weekly, Ending Tuesday
+            wem = Weekly, Ending Monday
+            wesu = Weekly, Ending Sunday
+            wesa = Weekly, Ending Saturday
+            bwew = Biweekly, Ending Wednesday
+            bwem = Biweekly, Ending Monday
+        """,
+        json_schema_extra={
+            "choices": [
+                "a",
+                "q",
+                "m",
+                "w",
+                "wef",
+                "weth",
+                "wew",
+                "wetu",
+                "wem",
+                "wesu",
+                "wesa",
+                "bwew",
+                "bwem",
+            ]
+        },
+    )
+    aggregation_method: Union[None, Literal["avg", "sum", "eop"]] = Field(
+        default=None,
+        description="""
+        A key that indicates the aggregation method used for frequency aggregation.
+            avg = Average
+            sum = Sum
+            eop = End of Period
+        """,
+        json_schema_extra={"choices": ["avg", "sum", "eop"]},
+    )
+    transform: Union[
+        None, Literal["chg", "ch1", "pch", "pc1", "pca", "cch", "cca", "log"]
+    ] = Field(
+        default=None,
+        description="""
+        Transformation type
+            None = No transformation
+            chg = Change
+            ch1 = Change from Year Ago
+            pch = Percent Change
+            pc1 = Percent Change from Year Ago
+            pca = Compounded Annual Rate of Change
+            cch = Continuously Compounded Rate of Change
+            cca = Continuously Compounded Annual Rate of Change
+            log = Natural Log
+        """,
+        json_schema_extra={
+            "choices": ["chg", "ch1", "pch", "pc1", "pca", "cch", "cca", "log"]
+        },
+    )
+    effr_only: bool = Field(
+        default=False,
+        description="Return data without quantiles, target ranges, and volume.",
+    )
+
+
+class FredFederalFundsRateData(FederalFundsRateData):
+    """FRED Federal Funds Rate Data."""
+
+
+class FredFederalFundsRateFetcher(
+    Fetcher[FredFederalFundsRateQueryParams, List[FredFederalFundsRateData]]
+):
+    """FRED Federal Funds Rate Fetcher."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FredFederalFundsRateQueryParams:
+        """Transform query."""
+        transformed_params = params.copy()
+        now = datetime.now().date()
+        if params.get("start_date") is None:
+            transformed_params["start_date"] = (
+                datetime(2016, 1, 1).date()
+                if params.get("effr_only") is False
+                else None
+            )
+        if params.get("end_date") is None:
+            transformed_params["end_date"] = now
+        return FredFederalFundsRateQueryParams(**transformed_params)
+
+        return FredFederalFundsRateQueryParams(**params)
```

Comment:
why do we have a second return statement here?

Reply:
Good question, could be merge conflict thing, but I find it amazing that the linters are not complaining about it.